shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -309,19 +309,38 @@ private MotionEvent toMotionEvent(float density, PlatformViewsChannel.PlatformVi
     MotionEventTracker.MotionEventId motionEventId =
         MotionEventTracker.MotionEventId.from(touch.motionEventId);
     MotionEvent trackedEvent = motionEventTracker.pop(motionEventId);
-    if (trackedEvent != null) {
-      return trackedEvent;
-    }
 
-    // TODO (kaushikiska) : warn that we are potentially using an untracked
-    // event in the platform views.
+    // Pointer coordinates in the tracked events are global to FlutterView
+    // framework converts them to be local to a widget, given that
+    // motion events operate on local coords, we need to replace these in the tracked
+    // event with their local counterparts.
     PointerProperties[] pointerProperties =
         parsePointerPropertiesList(touch.rawPointerPropertiesList)
             .toArray(new PointerProperties[touch.pointerCount]);
     PointerCoords[] pointerCoords =
         parsePointerCoordsList(touch.rawPointerCoords, density)
             .toArray(new PointerCoords[touch.pointerCount]);
 
+    if (trackedEvent != null) {
+      return MotionEvent.obtain(
+          trackedEvent.getDownTime(),
+          trackedEvent.getEventTime(),
+          trackedEvent.getAction(),
+          touch.pointerCount,
```

Comment:
with multiple touch, this pointerCount may unmatched with the trackedEvent.getAction() and cause some crash.
for example.
trackedEvent.getAction() -> 0x0106 (ActionIndex = 1, ACITON_POINTER_UP)
but pointerCount from flutter engine missed move event with the pointerUp, pointerCount will be 1